### ISSUE TITLE
Does this work ?

### ISSUE THREAD

**Issue by zabojad** [haxe-react/hxjs-react-redux#1]
Hi there !

I'm trying your macro for `haxe-redux` but it doesn't seem to work at my end:

```
@:build(react.redux.Macro.connect(PivotGridImpl))
class PivotGrid {

	public static function mapStateToProps(st : myproj.model.State, ownProps : PivotGridProps) : PivotGridProps {

		return {
			data: st.stat.data.data,
			configs: st.stat.config.pivotGrid.configs,
			currentConfigs: st.stat.config.pivotGrid.currentConfigs
		};
	}

	public static function mapDispatchToProps(dispatch : myproj.model.Thunk.Dispatch, ownProps : PivotGridProps) : PivotGridProps {

		return {
			fetchCmpData: function() { dispatch(StatCtrl.fetchPivotGridData(ownProps.cmp)); }
		};
	}
}

class PivotGridImpl extends ReactComponentOf<PivotGridProps, Dynamic, Dynamic> {
// ...
```

gives:
```
$hxClasses["myproj.view.cmp.stat.PivotGrid"] = myproj_view_cmp_stat_PivotGrid;
myproj_view_cmp_stat_PivotGrid.__name__ = true;
myproj_view_cmp_stat_PivotGrid.mapStateToProps = function(st,ownProps) {
	return { data : st.stat.data.data, configs : st.stat.config.pivotGrid.configs, currentConfigs : st.stat.config.pivotGrid.currentConfigs};
};
myproj_view_cmp_stat_PivotGrid.mapDispatchToProps = function(dispatch,ownProps) {
	return { fetchCmpData : function() {
		var tmp = myproj_ctrl_StatCtrl.fetchPivotGridData(ownProps.cmp);
		dispatch(tmp);
	}};
};
```
As it is not a React component anymore, it crashes at runtime:
```
Uncaught Error: myproj_view_cmp_stat_PivotGrid(...): A valid React element (or null) must be returned. You may have returned undefined, an array or some other invalid object.
```

Did I miss something ?

**Comment by zabojad** [comment]
I see the macro uses `__init__` which might be the cause of my problem as I am using also [haxe-loader](https://github.com/jasononeil/webpack-haxe-loader)...

**Comment by zabojad** [comment]
@kevinresol actually, that was not the main reason of my problem.

In my application code, I resolve dynamically the class of some of the component I render. And for this to work, the macro should also update the value of `$hxClasses["my.class.full.path"]` to the newly created comp...

@kevinresol what do you think about updating `hxjs-react-redux` with:
* updating `$hxClasses["my.class.full.path"]` along with the `my_class_full_path` var,
* maybe add an option to make the generated `__init__` method calling a static method of our component to avoid [the `haxe-modular` known issue with `__init__`](https://github.com/elsassph/haxe-modular#problem-with-init).

What do you think about that? Do you accept PRs?

Best,

Thomas

**Comment by kevinresol** [comment]
> updating $hxClasses["my.class.full.path"] along with the my_class_full_path var,

That should work



> maybe add an option to make the generated __init__ method calling a static method of our component

That should also work

Anyway, feel free to fork/PR. Also, you can have a look at another extern library https://github.com/elsassph/haxe-redux

**Comment by zabojad** [comment]
Yes, I've checked https://github.com/elsassph/haxe-redux before and it's to far away from the original react redux API to be safe to use in my case.

I just need a macro that replace the container class type with the result of the `connect()` call so `hxjs-react-redux` will dot it more...

OK thanks, I'll try to PR something soon...

**Comment by kevinresol** [comment]
@zabojad, this library itself is pretty rough and not of production quality (I used it in a really quickly done, finish and forget project). And I am pretty sure I am not going to maintain it, because I don't use redux now but [coconut](https://github.com/MVCoconut).
If you wish to maintain it, please kindly do so. Otherwise I would like to mark it as obsolete or something in the readme.

**Comment by zabojad** [comment]
@kevinresol Yes I'll maintain it. I have few long term projects based on Redux and will still use it for some time. This lib is just an extern anyway and the Redux API almost never changes... Let's keep it alive, maintaining it won't take me a lot of time.

**Comment by zabojad** [comment]
@kevinresol I've changed my mind since last message. I now use `https://github.com/elsassph/haxe-redux` so I guess we could mark this as obsolete in the README...
